tc/core/polyhedral/cuda/codegen.cc

```diff
@@ -681,6 +697,24 @@ size_t& nAstNodes() {
   return n;
 }
 
+// Collect ids of tensors that are only read on the Scop.
+std::unordered_set<isl::id, isl::IslIdIslHash> gatherReadOnlySet(
+    const MappedScop& mscop) {
+  auto root = mscop.schedule();
+  const auto& scop = mscop.scop();
+  std::unordered_set<isl::id, isl::IslIdIslHash> readOnlySet;
+
+  if (mscop.useReadOnlyCache) {
+    auto readOnlyAcceses =
+        scop.reads.universe().range().subtract(scop.writes.universe().range());
```

Comment:
These are not accesses, so the variable name is not appropriate.
I'd also break it up a bit to make it easier to read.
```
auto read = scop.reads.universe().range();
auto written = scop.writes.universe().range();
auto readOnlyAcceses = read.subtract(written);
```